src/index.js

```diff
@@ -0,0 +1,223 @@
+      params: {
+        q: cityInputValue,
+      },
+    })
+    .then((response) => {
+      console.log("search location success!");
+      state.lat = response.data[0].lat;
+      state.lon = response.data[0].lon;
+      state.city = cityInputValue;
+      searchTemperature();
+    })
+    .catch((error) => {
+      console.log("searchLocation error: " + error.response);
+    });
+};
+
+const searchTemperature = () => {
+  axios
+    .get("http://127.0.0.1:5000/weather", {
+      params: {
+        lat: state.lat,
+        lon: state.lon,
+      },
+    })
+    .then((response) => {
+      console.log("searchTemperature success!");
+      console.log(response.data);
+      state.tempCount = ((response.data.main.temp - 273.15) * 1.8 + 32).toFixed(
+        0
+      );
+      // return current temp and assign it to  temp count
+      const tempCount = document.getElementById("tempValue");
+      tempCount.textContent = `${state.tempCount}`;
+      tempRange();
+    })
+    .catch((error) => {
+      console.log("searchTemperature failing here!");
+      console.log(error);
+      console.log("searchTemperature error: " + error.response);
+    });
+};
+
+// confirmed this is working with console.log statement
+const handleCityNameSubmit = (event) => {
+  // this prevents the default event response / action from happening, enabling us to manage this independently
+  event.preventDefault();
+  const cityInputValue = event.target[0].value;
+  // call searchLocation function and passed cityInputValue as param
+  searchLocation(cityInputValue);
+};
+
+const registerEventHandlers = () => {
+  // assigns upTempArrow to that specific button. Document keyword grabs everything in index.html
+  // then you run func on document called querySelector. Then you are grabbing the id of increaseTempControl
+  // Do not need # in front of decreaseTempControl bc .getElementById does it automatically
+  const downTempArrow = document.getElementById("decreaseTempControl");
+  downTempArrow.addEventListener("click", decreaseTemperature);
+
+  const upTempArrow = document.getElementById("increaseTempControl");
+  upTempArrow.addEventListener("click", increaseTemperature);
+
+  const skySelector = document.getElementById("skySelect");
+  skySelector.addEventListener("change", updateSky);
+
+  updateCity();
+
+  const updateCityName = document.getElementById("cityNameInput");
+  updateCityName.addEventListener("input", updateCity);
+
+  const resetCity = document.getElementById("cityNameReset");
+  resetCity.addEventListener("click", resetCityName);
+
+  // submit events communicates all form elements
+  // vs a click event only grabs the element that was clicked on
+  // NEED submit event for communication across input field and submit button
+  const cityForm = document.getElementById("cityForm");
+  cityForm.addEventListener("submit", (event) => handleCityNameSubmit(event));
+};
+
+// calls function here to register events
+registerEventHandlers();
```

Comment:
Yes! To make this extra bug proof, we could ask JS to hold off on registering event handlers until the HTML and CSS finishes loading via listening to the `"DOMContentLoaded"`event, _then_ attaching event listeners to the HTML elements:
```
document.addEventListener("DOMContentLoaded", registerEventHandlers);
```